src/NzbDrone.Core/Datastore/Migration/031_add_artistmetadataid_constraint.cs

```diff
@@ -0,0 +1,25 @@
+using FluentMigrator;
+using NzbDrone.Core.Datastore.Migration.Framework;
+
+namespace NzbDrone.Core.Datastore.Migration
+{
+    [Migration(31)]
+    public class add_artistmetadataid_constraint : NzbDroneMigrationBase
+    {
+        protected override void MainDbUpgrade()
+        {
+            // Remove any duplicate artists
+            Execute.Sql(@"DELETE FROM Artists
+                          WHERE Id NOT IN (
+                            SELECT MIN(Artists.id) from Artists
```

Comment:
Should we delete the last added instead of the first added (smallest Id) to preserve the original added date?

Reply:
That's what it does :) it's deleting all ids that *aren't* the minimum when grouped by metadataid.

Reply:
Yea it's picking the min, sorry I was beat last night...

Reply:
@Qstick are you happy with this one now? Once it's in I can rebase the other series